collector/pg_stat_activity_summary.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const statActivitySummarySubsystem = "stat_activity_summary"
+
+func init() {
+	registerCollector(statActivitySummarySubsystem, defaultEnabled, NewPGStatActivitySummaryCollector)
+}
+
+type PGStatActivitySummaryCollector struct {
+	log log.Logger
+}
+
+func NewPGStatActivitySummaryCollector(config collectorConfig) (Collector, error) {
+	return &PGStatActivitySummaryCollector{log: config.logger}, nil
+}
+
+var (
+	statActivitySummaryActiveCount = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, statActivitySummarySubsystem, "active_count"),
+		"Number of active queries at time of sample",
+		[]string{"usename", "application", "endpoint", "command", "state", "wait_event", "wait_event_type"},
+		prometheus.Labels{},
+	)
+	statActivitySummaryMaxTxAgeInSeconds = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, statActivitySummarySubsystem, "max_tx_age_in_seconds"),
+		"Number of active queries at time of sample",
+		[]string{"usename", "application", "endpoint", "command", "state", "wait_event", "wait_event_type"},
+		prometheus.Labels{},
+	)
+
+	statActivitySummaryQuery = `
+	SELECT
+		usename AS usename,
+		a.matches[1] AS application,
+		a.matches[2] AS endpoint,
+		a.matches[3] AS command,
+		a.state AS state,
+		a.wait_event AS wait_event,
+		a.wait_event_type AS wait_event_type,
+		COUNT(*) active_count,
+		MAX(age_in_seconds) AS max_tx_age_in_seconds
+	FROM (
+		SELECT
+			usename,
+			regexp_matches(query, '^\s*(?:\/\*(?:application:(\w+),?)?(?:correlation_id:\w+,?)?(?:jid:\w+,?)?(?:endpoint_id:([\w/\-\.:\#\s]+),?)?.*?\*\/)?\s*(\w+)') AS matches,
+			state,
+			wait_event,
+			wait_event_type,
+			EXTRACT(EPOCH FROM (clock_timestamp() - xact_start)) AS age_in_seconds
```

Comment:
Same here, let's not compute the age, but expose the timestamp.
```suggestion
			EXTRACT(EPOCH FROM xact_start) AS timestamp_seconds
```